### ISSUE TITLE
Unable to authenticate local user using publickey

### ISSUE THREAD

**Issue by altaurog** [PowerShell/Win32-OpenSSH#136]
This may be a duplicate issue, but I don't know enough to tell.  I have Windows 7 running in a virtualbox vm on a debian 8 host.  The windows image is a free 32 bit testing image from MS, created using [ievms](https://github.com/xdissent/ievms).  The vbox configuration uses NAT with a forward from TCP port 2200 on the host to port 22 on the guest.  I have OpenSSH-Win32 installed on the guest.  I manually copied the x86 lsa dll for key authentication and checked that it is registered.  It works with password authentication, but not with publickey authentication.  I put the user's public key in `C:\Users\IEUser\.ssh\authorized_keys`.  I am not sure what, if any, permissions I need to change on filesystem objects.

My sshd_config file:
```
LogLevel DEBUG
StrictModes no
PubkeyAuthentication yes
AuthorizedKeysFile	.ssh/authorized_keys
PasswordAuthentication no
Subsystem	sftp	/win32openssh/bin/sftp-server.exe
Subsystem	scp	/win32openssh/bin/scp.exe
PubkeyAcceptedKeyTypes ssh-ed25519*,ssh-rsa*,ssh-dss*,ecdsa-sha2*
```

I tried connecting with the following command:
`ssh -p 2200 -l IEUser -i ~/.ssh/w7_id -o IdentitiesOnly=yes localhost`

Here is what I see in the logs:
```
4080 01:23:55 088 debug1: allocating new sfd, sfd [3] fd [4] handle [224] type [2]
4080 01:23:55 088 Connection from 10.0.2.2 port 51963 on 10.0.2.15 port 22
4080 01:23:55 088 debug1: Client protocol version 2.0; client software version OpenSSH_6.7p1 Debian-5+deb8u1
4080 01:23:55 088 debug1: match: OpenSSH_6.7p1 Debian-5+deb8u1 pat OpenSSH* compat 0x04000000
4080 01:23:55 088 debug1: Enabling compatibility mode for protocol 2.0
4080 01:23:55 088 debug1: Local version string SSH-2.0-OpenSSH_7.1p1 Microsoft_Win32_port_with_VS
4080 01:23:55 088 debug1: list_hostkey_types: ssh-rsa,ecdsa-sha2-nistp256,ssh-ed25519
4080 01:23:55 088 debug1: SSH2_MSG_KEXINIT sent
4080 01:23:55 088 debug1: SSH2_MSG_KEXINIT received
4080 01:23:55 088 debug1: kex: client->server aes128-ctr umac-64-etm@openssh.com none
4080 01:23:55 088 debug1: kex: server->client aes128-ctr umac-64-etm@openssh.com none
4080 01:23:55 088 debug1: expecting SSH2_MSG_KEX_ECDH_INIT
4080 01:23:55 088 debug1: SSH2_MSG_NEWKEYS sent
4080 01:23:55 088 debug1: expecting SSH2_MSG_NEWKEYS
4080 01:23:55 103 debug1: SSH2_MSG_NEWKEYS received
4080 01:23:55 103 debug1: KEX done
4080 01:23:55 135 debug1: userauth-request for user IEUser service ssh-connection method none
4080 01:23:55 135 debug1: attempt 0 failures 0
4080 01:23:55 135 Failed none for IEUser from 10.0.2.2 port 51963 ssh2
4080 01:23:55 135 debug1: userauth-request for user IEUser service ssh-connection method publickey
4080 01:23:55 135 debug1: attempt 1 failures 0
4080 01:23:55 135 debug1: test whether pkalg/pkblob are acceptable
4080 01:23:55 135 Postponed publickey for IEUser from 10.0.2.2 port 51963 ssh2
4080 01:23:55 135 debug1: userauth-request for user IEUser service ssh-connection method publickey
4080 01:23:55 135 debug1: attempt 2 failures 0
4080 01:23:55 135 debug1: -> LsaLogon()...
4080 01:23:55 135 debug1: Checking args...
4080 01:23:55 135 debug1: Setting up LSA Strings...
4080 01:23:55 135 debug1: LsaRegisterLogonProcess()...
4080 01:23:55 135 debug1: Retrieving Authentification Package ID...
4080 01:23:55 135 debug1: Allocating LsaAuth struct...
4080 01:23:55 135 debug1: Setting up TOKEN_SOURCE...
4080 01:23:55 135 debug1: Login attemp...
4080 01:23:55 150 debug1: SSH-LSA authorization failed. (err = 0, ntStat = c000006d).
4080 01:23:55 150 debug1: <- LsaLogon()...
4080 01:23:55 150 Failed publickey for IEUser from 10.0.2.2 port 51963 ssh2: ECDSA SHA256:LLpBzrejRha6sis9nt7Rq+f587FL8+UH5THZ8QsyGos
4080 01:23:55 150 debug1: userauth-request for user IEUser service ssh-connection method keyboard-interactive
4080 01:23:55 150 debug1: attempt 3 failures 1
4080 01:23:55 150 debug1: keyboard-interactive devs 
4080 01:23:55 150 debug1: auth2_challenge: user=IEUser devs=
4080 01:23:55 150 debug1: kbdint_alloc: devices ''
4080 01:23:55 150 Failed keyboard-interactive for IEUser from 10.0.2.2 port 51963 ssh2
4080 01:23:55 150 Connection closed by 10.0.2.2
4080 01:23:55 150 debug1: do_cleanup
```

I have tried both old and new rsa keys as well as an ecdsa key (as above).  I am not sure what else to try.

**Comment by manojampalam** [comment]
I'll try it out and let you know.

**Comment by an0t8** [comment]
I am having the same issue. x64 Windows 7 install.  Also using a local user, however other users are domain logins.

**Comment by wkwood** [comment]
Same problem however I can log in to a local account on the server using a password. 

3720 13:19:11 890 debug2: input_userauth_request: try method publickey
3720 13:19:11 890 debug1: test whether pkalg/pkblob are acceptable
3720 13:19:11 890 debug2: userauth_pubkey: authenticated 0 pkalg ssh-rsa
3720 13:19:11 890 Postponed publickey for testssh from 10.24.23.173 port 52346 ssh2
3720 13:19:15 201 debug1: userauth-request for user testssh service ssh-connection method publickey
3720 13:19:15 201 debug1: attempt 2 failures 0
3720 13:19:15 201 debug2: input_userauth_request: try method publickey
3720 13:19:15 201 debug1: -> LsaLogon()...
3720 13:19:15 201 debug1: Checking args...
3720 13:19:15 201 debug1: Setting up LSA Strings...
3720 13:19:15 201 debug1: LsaRegisterLogonProcess()...
3720 13:19:15 201 debug1: Retrieving Authentification Package ID...
3720 13:19:15 201 debug1: Allocating LsaAuth struct...
3720 13:19:15 201 debug3: Checking args...
3720 13:19:15 201 debug3: Adding authorized file [.ssh/authorized_keys] to LsaAuth...
3720 13:19:15 201 debug3: Computing total size of LsaAuth...
3720 13:19:15 201 debug3: Allocating new LsaAuth structure...
3720 13:19:15 201 debug3: Filling up LsaAuth struct...
3720 13:19:15 201 debug3: Converting [.ssh/authorized_keys] to UTF8...
3720 13:19:15 201 debug1: Setting up TOKEN_SOURCE...
3720 13:19:15 201 debug1: Login attemp...
3720 13:19:19 951 debug1: SSH-LSA authorization failed. (err = 0, ntStat = c000006d).
3720 13:19:19 951 debug1: <- LsaLogon()...
3720 13:19:19 951 debug2: userauth_pubkey: authenticated 0 pkalg ssh-rsa
3720 13:19:19 951 Failed publickey for testssh from 10.24.23.173 port 52346 ssh2: RSA SHA256:Dg7Akpk1PCiaw2BEGiz2wm5Ng4z+WwA+hJTzj3kg2EU
3720 13:19:19 951 debug3: userauth_finish: failure partial=0 next methods="publickey,password,keyboard-interactive"
3720 13:19:19 951 debug1: userauth-request for user testssh service ssh-connection method keyboard-interactive
3720 13:19:19 951 debug1: attempt 3 failures 1

**Comment by manojampalam** [comment]
Can you try the latest build? I just verified on Win7 SP1 x64 and key based auth using RSA keys works just fine.

**Comment by manojampalam** [comment]
Feel free to open if it doesn't work

**Comment by altaurog** [comment]
Same behavior here:

```
3216 12:46:25 580 Connection from 10.0.2.2 port 43222 on 10.0.2.15 port 22
3216 12:46:25 580 debug1: Client protocol version 2.0; client software version OpenSSH_6.7p1 Debian-5+deb8u1
3216 12:46:25 580 debug1: match: OpenSSH_6.7p1 Debian-5+deb8u1 pat OpenSSH* compat 0x04000000
3216 12:46:25 580 debug1: Enabling compatibility mode for protocol 2.0
3216 12:46:25 580 debug1: Local version string SSH-2.0-OpenSSH_7.1p1 Microsoft_Win32_port_with_VS
3216 12:46:25 580 debug1: list_hostkey_types: ssh-rsa,ecdsa-sha2-nistp256,ssh-ed25519
3216 12:46:25 580 debug1: SSH2_MSG_KEXINIT sent
3216 12:46:25 580 debug1: SSH2_MSG_KEXINIT received
3216 12:46:25 580 debug1: kex: algorithm: curve25519-sha256@libssh.org
3216 12:46:25 580 debug1: kex: host key algorithm: ecdsa-sha2-nistp256
3216 12:46:25 580 debug1: kex: client->server cipher: aes128-ctr MAC: umac-64-etm@openssh.com compression: none
3216 12:46:25 580 debug1: kex: server->client cipher: aes128-ctr MAC: umac-64-etm@openssh.com compression: none
3216 12:46:25 580 debug1: expecting SSH2_MSG_KEX_ECDH_INIT
3216 12:46:25 580 debug1: SSH2_MSG_NEWKEYS sent
3216 12:46:25 580 debug1: expecting SSH2_MSG_NEWKEYS
3216 12:46:25 580 debug1: SSH2_MSG_NEWKEYS received
3216 12:46:25 580 debug1: KEX done
3216 12:46:25 626 debug1: userauth-request for user IEUser service ssh-connection method none
3216 12:46:25 626 debug1: attempt 0 failures 0
3216 12:46:25 626 Failed none for IEUser from 10.0.2.2 port 43222 ssh2
3216 12:46:25 626 debug1: userauth-request for user IEUser service ssh-connection method publickey
3216 12:46:25 626 debug1: attempt 1 failures 0
3216 12:46:25 626 debug1: test whether pkalg/pkblob are acceptable
3216 12:46:25 626 Postponed publickey for IEUser from 10.0.2.2 port 43222 ssh2
3216 12:46:25 626 debug1: userauth-request for user IEUser service ssh-connection method publickey
3216 12:46:25 626 debug1: attempt 2 failures 0
3216 12:46:25 626 debug1: -> LsaLogon()...
3216 12:46:25 626 debug1: Checking args...
3216 12:46:25 626 debug1: Setting up LSA Strings...
3216 12:46:25 626 debug1: LsaRegisterLogonProcess()...
3216 12:46:25 626 debug1: Retrieving Authentification Package ID...
3216 12:46:25 626 debug1: Allocating LsaAuth struct...
3216 12:46:25 626 debug1: Setting up TOKEN_SOURCE...
3216 12:46:25 626 debug1: Login attemp...
3216 12:46:25 642 debug1: SSH-LSA authorization failed. (err = 0, ntStat = c000006d).
3216 12:46:25 642 debug1: <- LsaLogon()...
3216 12:46:25 642 Failed publickey for IEUser from 10.0.2.2 port 43222 ssh2: ECDSA SHA256:LLpBzrejRha6sis9nt7Rq+f587FL8+UH5THZ8QsyGos
3216 12:46:25 642 debug1: userauth-request for user IEUser service ssh-connection method keyboard-interactive
3216 12:46:25 642 debug1: attempt 3 failures 1
3216 12:46:25 642 debug1: keyboard-interactive devs 
3216 12:46:25 642 debug1: auth2_challenge: user=IEUser devs=
3216 12:46:25 642 debug1: kbdint_alloc: devices ''
3216 12:46:25 642 Failed keyboard-interactive for IEUser from 10.0.2.2 port 43222 ssh2
3216 12:46:25 642 Connection closed by 10.0.2.2
3216 12:46:25 642 debug1: do_cleanup
```

Is there something else I should try/check?
